producer/src/blockchain/blockchain.go

```diff
@@ -258,3 +258,84 @@ func GetBlockByHash(hash []byte, filename string, database string) ([]byte, erro
 
 	return bl, nil
 }
+
+/*
+Takes in the ledger file's name and the desired name for the metadata file
+Creates the database if it doesn't exist, inserts the appropriate column names
+Moves through the ledger file and extracts blocks, repopulating the metadata file
+Test will check to make sure a ledger populated by AddBlock() will result in the same
+metadata table as would result from here
+Return an error if anything I/O operations fail with detailed message
+*/
+func RecoverBlockchainMetadata(ledgerFilename string, metadataFilename string) error {
+	_, err := os.Stat(metadataFilename)
+	if err != nil {
+		// create database
+		_, err = os.Create(metadataFilename)
+
+		// open database
+		db, err := sql.Open("sqlite3", "testDatabase.db")
```

Comment:
`"testDatabase.db"` shouldn't exist, only whatever is passed in as `metadataFilename`

Reply:
Should I also change the string for `table` in unit test to `testDatabase.db`? Because otherwise when the unit test calls `GetBlockByHeight(n, testBlockchain.dat, testDatabase.db)`, it will definitely return an error.

![image](https://user-images.githubusercontent.com/42977312/55271874-4cd89800-5282-11e9-9091-0b21c88a2988.png)

Reply:
I see what you mean. A mistake on my part when I copy-pasted too liberally. For the "get" functions, pass in `blockchain` and `table` instead of the strings that are currently being passed in. That should fix it.